lib/client/connection.js

```diff
@@ -584,3 +600,47 @@ Connection.prototype._firstQuery = function(fn) {
     }
   }
 };
+
+Connection.prototype._firstSnapshotRequest = function (fn) {
+  for (var id in this.snapshotRequests) {
+    var snapshotRequest = this.snapshotRequests[id];
+    if (fn(snapshotRequest)) {
+      return snapshotRequest;
+    }
+  }
+};
+
+/**
+ * Fetch a read-only snapshot at a given version
+ *
+ * @param collection - the collection name of the snapshot
+ * @param id - the ID of the snapshot
+ * @param version (optional) - the version number to fetch
+ * @param callback - (error, snapshot) => void, where snapshot takes the following schema:
+ *
+ * {
+ *   id: string;         // ID of the snapshot
+ *   v: number;          // version number of the snapshot
+ *   type: any;          // the OT type of the snapshot, or null if it doesn't exist or is deleted
+ *   data: any;          // the snapshot
+ * }
+ *
+ */
+Connection.prototype.fetchSnapshot = function(collection, id, version, callback) {
+  if (typeof version === 'function') {
+    callback = version;
+    version = null;
+  }
+
+  var snapshotRequest = new SnapshotRequest(this, collection, id, version, callback);
+  snapshotRequest.version = version;
```

Comment:
This line is probably redundant, considering that `version` is passed into the constructor just above.

Reply:
Good catch.